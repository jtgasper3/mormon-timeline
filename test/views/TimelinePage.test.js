import { waitFor } from "@testing-library/vue";
import { describe, it, expect, vi } from "vitest";
import TimelinePage from "@/views/TimelinePage.vue";
import { renderWithVuetify } from "../vitest.vuetifySetup.js";

vi.mock("@/services", () => {
  return {
    getEvents: () => [
      {
        id: "bom_bob_birth",
        title: "Mormon Timeline Created",
        date: new Date("2022-04-22T00:00:00"),
        description: "This is a test",
        references: [],
        category: "bom",
        tags: ["test"],
      },
    ],
  };
});

describe("TimelinePage", () => {
  it("mount component", async () => {
    const { html } = renderWithVuetify(TimelinePage);

    await waitFor(() => {});
    expect(html()).toMatchSnapshot();
  });
});
