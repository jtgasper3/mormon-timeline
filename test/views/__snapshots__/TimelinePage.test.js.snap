// Vitest Snapshot v1

exports[`TimelinePage > mount component 1`] = `
"<div class=\\"v-row\\">
  <div class=\\"v-col\\">
    <div class=\\"v-input v-input--horizontal v-input--density-default v-text-field\\">
      <!---->
      <div class=\\"v-input__control\\">
        <div class=\\"v-field v-field--prepended v-field--has-label v-field--variant-contained v-theme--light\\" role=\\"textbox\\">
          <div class=\\"v-field__overlay\\"></div>
          <div class=\\"v-field__loader\\">
            <div class=\\"v-progress-linear v-theme--light\\" style=\\"--v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
              <!---->
              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
              <transition-stub>
                <div class=\\"v-progress-linear__indeterminate\\">
                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <div class=\\"v-field__prepend-inner\\"><i class=\\"mdi-magnify mdi v-icon notranslate v-icon--size-default v-theme--light\\" aria-hidden=\\"true\\"></i>
            <!---->
          </div>
          <div class=\\"v-field__field\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\">
              <!---->Search
            </label><label class=\\"v-label v-field-label\\" for=\\"input-1\\">
              <!---->Search
            </label>
            <!---->
            <div class=\\"v-field__input\\">
              <!----><input size=\\"1\\" type=\\"text\\" id=\\"input-1\\" />
            </div>
            <!---->
          </div>
          <!---->
          <!---->
          <div class=\\"v-field__outline\\">
            <!---->
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <div class=\\"v-input__details\\">
        <transition-group-stub tag=\\"div\\" class=\\"v-messages\\">
          <!---->
        </transition-group-stub>
        <!---->
      </div>
    </div>
  </div>
</div>
<div class=\\"v-row\\">
  <div class=\\"v-col\\">
    <div class=\\"v-timeline v-timeline--vertical v-timeline--align-center v-theme--light v-timeline--density-default\\" style=\\"--v-timeline-line-thickness: 2px; --v-timeline-line-inset: 0px;\\">
      <div class=\\"v-timeline-item v-timeline-item--fill-dot\\" style=\\"--v-timeline-dot-size: 0px;\\">
        <div class=\\"v-timeline-item__body\\">
          <div>
            <h2 class=\\"mt-n1 headline font-weight-light mb-4 text-cyan justify-end\\">Mormon Timeline Created</h2>
            <!--v-if-->
            <div>This is a test</div>
          </div>
        </div>
        <div class=\\"v-timeline-divider v-timeline-divider--fill-dot\\">
          <div class=\\"v-timeline-divider__dot v-timeline-divider__dot--size-small\\">
            <div class=\\"v-timeline-divider__inner-dot bg-cyan\\">
              <!---->
            </div>
          </div>
          <div class=\\"v-timeline-divider__line\\"></div>
        </div>
        <div class=\\"v-timeline-item__opposite\\">
          <div class=\\"pt-1 headline font-weight-bold text-cyan\\">April 22nd, 2022</div>
        </div>
      </div>
    </div>
  </div>
</div>
<div class=\\"v-card v-card--flat v-theme--dark v-card--density-default v-card--fixed v-card--variant-contained px-1\\" style=\\"background-color: rgba(0, 0, 0, 0.1); width: 90px; top: 104px; right: -35px;\\">
  <!---->
  <div class=\\"v-card__underlay\\"></div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!----><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--dark text-primary v-btn--density-default v-btn--size-default v-btn--variant-plain\\" aria-label=\\"Filter\\" aria-haspopup=\\"dialog\\" aria-expanded=\\"false\\">
    <div class=\\"v-btn__overlay\\"></div>
    <div class=\\"v-btn__underlay\\"></div>
    <!----><i class=\\"mdi-filter-menu mdi v-icon notranslate v-icon--size-default text-primary v-theme--dark\\" aria-hidden=\\"true\\" dark=\\"\\" aria-describedby=\\"v-tooltip-2\\"></i>
    <!---->
  </button>
  <!--teleport start-->
  <!--teleport end-->
  <!---->
</div>
<!--teleport start-->
<!--teleport end-->"
`;
